src/main/java/nus/climods/logic/commands/AddCommand.java

```diff
@@ -38,19 +42,35 @@ public AddCommand(String moduleCode) {
         toAdd = moduleCode;
     }
 
+    private Module getModule(Model model, String moduleCode) throws CommandException {
+        Optional<Module> optionalModule = model.getModuleList().getListModule(moduleCode);
+
+        if (optionalModule.isEmpty()) {
+            throw new CommandException(MESSAGE_MODULE_NOT_FOUND);
+        }
+        return optionalModule.get();
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        UserModule moduleToAdd = new UserModule(toAdd, model);
+
+        UserModule moduleToAdd = new UserModule(getModule(model, toAdd));
```

Comment:
I think it is better to check on the command level instead. Personally I prefer the data model to be as separated from the verification/validation so that when any potential error comes from the execution of the command instead of during a initialization of a data object.

Reply:
@sarrrdin can't check inside UserModule now cos we r not passing the ModelManager to the UserModule due to the cyclical dependency that is created when we pass ModelManager to UserModule

Reply:
noted of it!